### ISSUE TITLE
The -auto_ext flag should vary based on the caller's package

### ISSUE THREAD

**Issue by srchulo** [srchulo/JSON-Slurper#3]
As pointed out in #2 by @tobyink, the `-auto_ext` flag currently has a global effect. Instead, we should make `-auto_ext` depend on the package of the caller.

**Comment by Grinnz** [comment]
If I may make an alternative suggestion, consider exporting different versions of the functions depending on the export flag. The package name of the caller is not always what will end up calling the function (re-exporting, calling in a coderef passed somewhere else, etc)

**Comment by srchulo** [comment]
@Grinnz that's a good suggestion. It seems like [Exporter](https://metacpan.org/pod/Exporter) doesn't support exporting subroutines under different names. Do you have any suggestions for an alternative that would allow this?

**Comment by Grinnz** [comment]
It's not easily doable with the default Exporter setup unfortunately. If you don't mind a non-core module, Exporter::Tiny can do it: https://metacpan.org/pod/Exporter::Tiny::Manual::QuickStart#Generators - otherwise you would need some complicated manual exporting probably.

**Comment by srchulo** [comment]
Yeah, I wanted to keep a core module, but the manual exporting doesn't sound worth it. Thanks for pointing me to that! I'll take a look.

**Comment by Grinnz** [comment]
It doesn't look like you can allow arbitrary import flags with Exporter::Tiny, but you *could* document something like `use JSON::Slurper { auto_ext => 1 }, qw(slurp_json);` and then that would be available in the `$globals` hash it talks about here: https://metacpan.org/pod/Exporter::Tiny::Manual::Exporting#More-on-Generators

**Comment by tobyink** [comment]
There's actually an (undocumented?) feature of Exporter::Tiny that if `$EXPORT_TAGS{auto_ext}` doesn't exist, then:

```
use JSON::Slurper qw(-auto_ext slurp_json);
```

Acts as a shortcut for:

```
use JSON::Slurper { auto_ext => 1 }, qw(slurp_json);
```

Example script:
```
use v5.16;

package My::Exporter {
	use Exporter::Shiny qw(bar);
	sub _generate_bar {
		use Data::Dumper;
		warn Dumper \@_;
		return sub { say "bar was called" };
	}
};

use My::Exporter qw( -foo bar );
```

**Comment by srchulo** [comment]
@tobyink that's very helpful! Thanks :)

**Comment by srchulo** [comment]
Thank you both for the help and advice! This was fixed in #4 by 36c17e6. I think this is much better now 👍
